src/include/settings/settings_defs.h

```diff
@@ -276,3 +284,37 @@ SETTING_string(
     true,
     terrier::settings::Callbacks::NoOp
 )
+
+SETTING_bool(
+    override_num_threads,
+    "Overrides the number of threads to use (default: false)",
+    false,
+    true,
+    terrier::settings::Callbacks::NoOp
+)
+
+SETTING_int(
+    num_threads,
+    "Number of threads for parallel query execution (default: 1)",
+    1,
+    1,
+    128,
+    true,
+    terrier::settings::Callbacks::NoOp
+)
+
+SETTING_bool(
+    use_counters,
+    "Whether to use counters (default: false)",
+    false,
+    true,
+    terrier::settings::Callbacks::NoOp
+)
+
+SETTING_bool(
+    use_pipeline_metrics,
```

Comment:
Can you consolidate this with `metrics_pipeline`? Also I think we can rename this to `pipeline_metrics_enable` to be consistent with our other flag namings. Also, please apply the naming to other metrics, e.g., `logging_metrics_enable`, and `counters_enable` as well.